### ISSUE TITLE
Support Newsletter Feedbacks

### ISSUE THREAD

**Issue by ItzNotABug** [ItzNotABug/ghosler#10]
Building on the discussion in [issue #2](https://github.com/ItzNotABug/ghosler/issues/2#issuecomment-1854273005), I've reconsidered the approach to managing newsletter feedback. Initially, I was confused about data consistency when saving feedback both in `Ghost` and `Ghosler`. However, a more streamlined solution can be used.

`Ghost` conveniently  collects newsletter feedback using a specific URL format. This allows us to directly access feedback and sentiment data through the Ghost Admin API.

#### URL format - 
```
https://{siteUrl}/#/feedback/{postID}/{feedbackType}/?uuid={memberUUID}
```
Here, `feedbackType` can be `0` (More like this) or `1` (Less like this).

We should focus on these fields in our API requests: - 
- `sentiment`,
- `count.positive_feedback`, 
- `count.negative_feedback`

---
### Flow
1. A member clicks a feedback option (e.g., "More like this").
2. They are redirected to the specified URL, with the appropriate feedbackType.
3. `Ghost` presents a confirmation popup or sheet to the member.
4. `Ghosler` retrieves this feedback data, using it for analytics and dashboard displays.

**Comment by ItzNotABug** [comment]
Hey @viriatusX 👋,
The support for newsletter feedback is also added now, currently in the [`feedback-support`](https://github.com/ItzNotABug/ghosler/tree/feedback-support) branch.

**Comment by ItzNotABug** [comment]
This `feedbackType` flag gave me a good confusion, multiple times.

Looking at the Ghost's impl. I see:
Like = 1,
Dislike = 0

Reference:
https://github.com/TryGhost/Ghost/blob/efb2b07c601cd557976bcbe12633f072da5c22a7/ghost/email-service/lib/EmailRenderer.js#L1000C9-L1009

**Comment by viriatusX** [comment]
Wow! Incredible!
But in my case not work the new branch :(

<img width="875" alt="Captura de pantalla 2023-12-21 a las 19 04 18" src="https://github.com/ItzNotABug/ghosler/assets/64467637/4304f939-56b2-4f5f-b9e8-b05277278776">

**Comment by viriatusX** [comment]
<img width="811" alt="Captura de pantalla 2023-12-21 a las 19 08 29" src="https://github.com/ItzNotABug/ghosler/assets/64467637/f360ddef-3140-475d-97be-3fe51db67ae6">

The new branch not work correctly (in my case). Look the error.
When I start Ghosler, I go to Settings, modify Newsletter Settings and it does not work "save" or anything else. Then it shows this error and the app stops working showing this error.

**Comment by ItzNotABug** [comment]
@viriatusX,
Unable to reproduce, can you post the contents of `ghosler logs error`?

**Comment by viriatusX** [comment]
I reinstalled it with "sudo" in my Ubuntu VM and everything is fine. But I don't receive the newsletters to my email. I have checked absolutely everything and it is correct. "Error logs" shows nothing and "Debug Logs" shows everything correctly. But the email does not arrive.
I have tried 2 different SMTP servers and nothing.

**Comment by ItzNotABug** [comment]
> I reinstalled it with "sudo" in my Ubuntu VM and everything is fine. But I don't receive the newsletters to my email. I have checked absolutely everything and it is correct. "Error logs" shows nothing and "Debug Logs" shows everything correctly. But the email does not arrive. I have tried 2 different SMTP servers and nothing.

I've identified a few issues myself. 
Would be fixed by tomorrow I guess, once the `webhook_support` branch is complete.

Thanks for the feedback.

**Comment by ItzNotABug** [comment]
Hey @viriatusX 👋,
The latest release (`0.6`) should work fine. 
Tested with a clean install, seems to be working fine now.

**Comment by viriatusX** [comment]
> Hey @viriatusX 👋, The latest release (`0.6`) should work fine. Tested with a clean install, seems to be working fine now.

Now all works perfectly! Thanks!!

Happy Christmas 🎄

**Comment by viriatusX** [comment]
Hello!
I have used Ghosler in real production for the first time. Many users do not receive the email. I attach the "error logs".


<img width="714" alt="Captura de pantalla 2023-12-24 a las 11 45 23" src="https://github.com/ItzNotABug/ghosler/assets/64467637/6c4abbcb-4a41-44d8-8ba4-e7ffe3af2354">

`[2023-12-24 10:28:36 UTC] => [ERROR] => Newsletter: Error: Data command failed: 421-4.3.0 Temporary System Problem. Try again later (10). For more information,
421-4.3.0 go to
421 4.3.0  https://support.google.com/a/answer/3221692 g3-20020a05600c310300b0040c6b2c8fa9sm21226207wmo.41 - gsmtp
at SMTPConnection._formatError (/home/juani/Escritorio/node_modules/nodemailer/lib/smtp-connection/index.js:790:19)
at SMTPConnection._actionDATA (/home/juani/Escritorio/node_modules/nodemailer/lib/smtp-connection/index.js:1678:34)
at SMTPConnection.<anonymous> (/home/juani/Escritorio/node_modules/nodemailer/lib/smtp-connection/index.js:1650:26)
at SMTPConnection._processResponse (/home/juani/Escritorio/node_modules/nodemailer/lib/smtp-connection/index.js:969:20)
at SMTPConnection._onData (/home/juani/Escritorio/node_modules/nodemailer/lib/smtp-connection/index.js:755:14)
at SMTPConnection._onSocketData (/home/juani/Escritorio/node_modules/nodemailer/lib/smtp-connection/index.js:193:44)
at TLSSocket.emit (node:events:513:28)
at addChunk (node:internal/streams/readable:324:12)
at readableAddChunk (node:internal/streams/readable:297:9)
at Readable.push (node:internal/streams/readable:234:10)`

**Comment by ItzNotABug** [comment]
@viriatusX Can you create a new issue to keep a better track?
